backend/experiment/static/experiment_form.js

```diff
@@ -60,3 +64,43 @@ function floatingSubmitRow() {
         }
     });
 }
+/**
+ * Re-initialize django-select2 on form change.
+ * This is needed because django-select2 is not re-initialized on new inlines.
+ * Let's check again in the future if this issue is fixed in django-select2 
+ * and remove this function.
+ * See also: https://github.com/codingjoe/django-select2/pull/300
+ * @returns {void}
+ */
+function reinitializeDjangoSelect2OnFormChange() {
+
+    const $ = django.jQuery;
+
+    return $(function () {
+        $('.django-select2').djangoSelect2()
+
+        // This part fixes new inlines not having the correct select2 widgets
+        function handleFormsetAdded(row, formsetName) {
+            // Converting to the "normal jQuery"
+            var jqRow = django.jQuery(row)
+
+            // Because select2 was already instantiated on the empty form, we need to remove it, destroy the instance,
+            // and re-instantiate it.
+            jqRow.find('.django-select2').parent().find('.select2-container').remove()
+            jqRow.find('.django-select2').djangoSelect2('destroy');
+            jqRow.find('.django-select2').djangoSelect2()
+        }
+
+        // See: https://docs.djangoproject.com/en/dev/ref/contrib/admin/javascript/#supporting-versions-of-django-older-than-4-1
+        django.jQuery(document).on('formset:added', function (event, $row, formsetName) {
+            if (event.detail && event.detail.formsetName) {
```

Comment:
Since we pinned the Django version, is this `if/else` necessary?

Reply:
In a normal world I would say no, but somehow the `formset:added` event is dispatched twice whenever adding a new block form, one in the >= 4.1 and one in the < 4.1 format... I don't know which one is the "good" one. Both seem to accomplish in initializing the select2 input correctly and having the function be called twice also does not seem to cause any harm. I was thinking of leaving it like this until the library releases a real fix.